### PR TITLE
Add docker compatibility

### PR DESCRIPTION
**Pull request by GabrielDS**
* Adds Dockerfile for image build.
* Adds dockercompose to facilitate workflow.
* Adds dockerignore for futures adjusts.
* Adds "--host" parameter on uvicorn to fix networking problem when run inside container. 

Notes:

The pipenv is slow down the image build, also, increase ~37mb of image's size but this pull request is just to add docker flow for development enviroment. To run on production enviroments safety, we will need to do somes ajusts that could change the project a little bit. 

The app waits a environment variable called "PORT" but, even without it, i can run the api using the port 8000 (a little bit strange). 

#252

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -0,0 +1,14 @@
+version: '3.7'
+
+services:
+  api:
+    build: .
+    image: expdev07/coronavirus-tracker-api:2.0
+    restart: always
+    command: pipenv --python 3.7 run dev
```
Comment (Kilo59) [inline docker-compose.yml:8]: We shouldn't be running the dev version with the "reload" flag in the container.

File: Pipfile [hunk 2]
```diff
@@ -26,7 +26,7 @@ uvicorn = "*"
 python_version = "3.8"
 
 [scripts]
-dev = "uvicorn app.main:APP --reload"
+dev = "uvicorn  --host 0.0.0.0 app.main:APP --reload"
```
Comment (Kilo59) [inline Pipfile:29]: You shouldn't be changing this.

File: docker-compose.yml (hunk 1 above)
Comment (GabrielDS) [reply docker-compose.yml]: Done!

File: Pipfile (hunk 2 above)
Comment (GabrielDS) [reply Pipfile]: Done!

File: Dockerfile [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+
+# PREPARE FOLDER
+WORKDIR /api
+
+# COPY DEPENDENCIES
+COPY Pipfile Pipfile.lock ./
+
+# INSTALL DEPENDENCIES
+RUN pip install pipenv
+RUN pipenv install --system --deploy --dev
```
Comment (Kilo59) [inline Dockerfile:15]: We should install from the requirements files once this goes in.  #245

Comment (GabrielDS) [reply Dockerfile:15]: Which requirements file? requirements-dev.txt or requirements.txt?

Comment (Kilo59) [reply Dockerfile:15]: Just `requirements.txt`

Comment (GabrielDS) [reply Dockerfile]: Done!!

**Pull request merged**